src/Locale.php

```diff
@@ -79,19 +80,21 @@ public function process(ServerRequestInterface $request, RequestHandlerInterface
         $uri = $request->getUri();
         $path = $uri->getPath();
         $query = $uri->getQuery();
-        $newPath = null;
         $locale = $this->getLocaleFromPath($path);
 
         if ($locale !== null) {
-            if ($request->getMethod() === Method::GET) {
-                $newPath = substr($path, strlen($locale) + 1) ?: '/';
+            if ($locale === $this->defaultLocale && $request->getMethod() === Method::GET) {
+                return $this->saveLocale(
+                    $locale,
+                    $this->createRedirectResponse(substr($path, strlen($locale) + 1) ?: '/', $query)
+                );
             }
         } else {
             /** @psalm-var array<string, string> $queryParameters */
             $queryParameters = $request->getQueryParams();
             $locale = $this->getLocaleFromQuery($queryParameters);
 
-            if ($locale === null) {
+            if ($locale === null && $this->cookieDuration !== null) {
```

Comment:
The previous code handles this case:

1. Locale was previously saved in cookies.
2. User wants to retrieve it from cookies.
3. User doesn't want to save it to cookies anymore.

With the new changes it will be saved anyway.

Maybe using a separate boolean flag would be still better / more explicit.

Reply:
It's decide developer, not user. If developer disable cookie, then we should not use cookies.

Reply:
If cookies are disabled, they should not be created, indeed.

Reply:
By user I mean developer here.

Reply:
I mean the same.